src/cursor/abstract_cursor.ts

```diff
@@ -0,0 +1,710 @@
+
+/** @internal */
+export interface AbstractCursorOptions extends BSONSerializeOptions {
+  session?: ClientSession;
+  readPreference?: ReadPreferenceLike;
+  batchSize?: number;
+  maxTimeMS?: number;
+  comment?: Document | string;
+  tailable?: boolean;
+  awaitData?: boolean;
+  noCursorTimeout?: boolean;
+}
+
+/** @internal */
+export type InternalAbstractCursorOptions = Omit<AbstractCursorOptions, 'readPreference'> & {
+  // resolved
+  readPreference: ReadPreference;
+
+  // cursor flags, some are deprecated
+  oplogReplay?: boolean;
+  exhaust?: boolean;
+  partial?: boolean;
+};
+
+/** @internal */
+export abstract class AbstractCursor extends EventEmitter {
+  [kId]?: Long;
+  [kSession]?: ClientSession;
+  [kServer]?: Server;
+  [kNamespace]: MongoDBNamespace;
+  [kDocuments]: Document[];
+  [kTopology]: Topology;
+  [kTransform]?: (doc: Document) => Document;
+  [kInitialized]: boolean;
+  [kClosed]: boolean;
+  [kKilled]: boolean;
+  [kOptions]: InternalAbstractCursorOptions;
+
+  /** @event */
+  static readonly CLOSE = 'close' as const;
+
+  constructor(
+    topology: Topology,
+    namespace: MongoDBNamespace,
+    options: AbstractCursorOptions = {}
+  ) {
+    super();
+
+    this[kTopology] = topology;
+    this[kNamespace] = namespace;
+    this[kDocuments] = []; // TODO: https://github.com/microsoft/TypeScript/issues/36230
+    this[kInitialized] = false;
+    this[kClosed] = false;
+    this[kKilled] = false;
+    this[kOptions] = {
+      readPreference:
+        options.readPreference && options.readPreference instanceof ReadPreference
+          ? options.readPreference
+          : ReadPreference.primary,
+      ...pluckBSONSerializeOptions(options)
+    };
+
+    if (typeof options.batchSize === 'number') {
+      this[kOptions].batchSize = options.batchSize;
+    }
+
+    if (typeof options.comment !== 'undefined') {
+      this[kOptions].comment = options.comment;
+    }
+
+    if (typeof options.maxTimeMS === 'number') {
+      this[kOptions].maxTimeMS = options.maxTimeMS;
+    }
+
+    if (options.session instanceof ClientSession) {
+      this[kSession] = options.session;
+    }
+  }
+
+  get id(): Long | undefined {
+    return this[kId];
+  }
+
+  get topology(): Topology {
+    return this[kTopology];
+  }
+
+  get server(): Server | undefined {
+    return this[kServer];
+  }
+
+  get namespace(): MongoDBNamespace {
+    return this[kNamespace];
+  }
+
+  get readPreference(): ReadPreference {
+    return this[kOptions].readPreference;
+  }
+
+  get session(): ClientSession | undefined {
+    return this[kSession];
+  }
+
+  /** @internal */
+  get cursorOptions(): InternalAbstractCursorOptions {
+    return this[kOptions];
+  }
+
+  get closed(): boolean {
+    return this[kClosed];
+  }
+
+  get killed(): boolean {
+    return this[kKilled];
+  }
+
+  // NOTE: should we remove these? They are currently needed by a number of tests
+  isClosed(): boolean {
+    return this.closed;
```

Comment:
```suggestion
    return this[kClosed];
```

I think this should be using the symbol, if we're keeping it.

Reply:
In terms of *whether* to keep it, I don't see how users get any utility out of it; I've vote to remove it if it doesn't take too much rewriting of tests.

Reply:
I updated this to remove the `isClosed` method on `AbstractCursor` and `ChangeStream`, instead depending on the `closed` accessor